homeassistant/components/flo/device.py

```diff
@@ -0,0 +1,174 @@
+
+    @property
+    def mac_address(self) -> str:
+        """Return ieee address for device."""
+        return self._mac_address
+
+    @property
+    def manufacturer(self) -> str:
+        """Return manufacturer for device."""
+        return self._manufacturer
+
+    @property
+    def model(self) -> str:
+        """Return model for device."""
+        return self._model
+
+    @property
+    def rssi(self) -> float:
+        """Return rssi for device."""
+        return self._rssi
+
+    @property
+    def last_heard_from_time(self) -> str:
+        """Return lastHeardFromTime for device."""
+        return self._last_heard_from_time
+
+    @property
+    def device_type(self) -> str:
+        """Return the device type for the device."""
+        return self._device_type
+
+    @property
+    def available(self) -> bool:
+        """Return True if device is available."""
+        return self._available
+
+    @property
+    def current_system_mode(self) -> str:
+        """Return the current system mode."""
+        return self._current_system_mode
+
+    @property
+    def target_system_mode(self) -> str:
+        """Return the target system mode."""
+        return self._target_system_mode
+
+    @property
+    def current_flow_rate(self) -> float:
+        """Return current flow rate in gpm."""
+        return self._current_flow_rate
+
+    @property
+    def current_psi(self) -> float:
+        """Return the current pressure in psi."""
+        return self._current_psi
+
+    @property
+    def temperature(self) -> float:
+        """Return the current temperature in degrees F."""
+        return self._temperature
+
+    @property
+    def consumption_today(self) -> float:
+        """Return the current consumption for today in gallons."""
+        return self._consumption_today
+
+    @property
+    def firmware_version(self) -> str:
+        """Return the firmware version for the device."""
+        return self._firmware_version
+
+    async def _update_device(self, *_) -> None:
+        """Update the device information from the API."""
+        device_information = await self.api_client.device.get_info(self._flo_device_id)
+        _LOGGER.debug("Flo device data: %s", device_information)
+        self._available = device_information["isConnected"]
+        self._firmware_version = device_information["fwVersion"]
+        self._rssi = device_information["connectivity"]["rssi"]
+        self._last_heard_from_time = device_information["lastHeardFromTime"]
+        self._current_system_mode = device_information["systemMode"]["lastKnown"]
+        self._target_system_mode = device_information["systemMode"]["target"]
+        self._current_flow_rate = device_information["telemetry"]["current"]["gpm"]
+        self._current_psi = device_information["telemetry"]["current"]["psi"]
+        self._temperature = device_information["telemetry"]["current"]["tempF"]
+        self.hass.bus.async_fire(f"{self.mac_address}-{SIGNAL_DEVICE_UPDATED}")
```

Comment:
You should use the dispatcher ? or have people subscribe to the data update coordinator directly.

Reply:
gonna swap to the coordinator so will have them subscribe to that

Reply:
swapped to DataUpdateCoordinator